modules/core/include/opencv2/core/utility.hpp

```diff
@@ -513,30 +513,36 @@ class CV_EXPORTS AutoLock
     AutoLock& operator = (const AutoLock&);
 };
 
+// TLS interface
 class CV_EXPORTS TLSDataContainer
 {
-private:
-    int key_;
 protected:
-    TLSDataContainer();
-    virtual ~TLSDataContainer();
-public:
-    virtual void* createDataInstance() const = 0;
-    virtual void deleteDataInstance(void* data) const = 0;
+    TLSDataContainer() { key = (size_t)-1; };
+    virtual ~TLSDataContainer() {};
 
+    void  getKey();
     void* getData() const;
+    void  releaseKey();
+
+private:
+    virtual void* createDataInstance() const = 0;
+    virtual void  deleteDataInstance(void* pData) const = 0;
+
+    mutable size_t key;
```

Comment:
ABI check fails here, need to leave `int key_;` ("mutable" can be added without any ABI problems)